src/glaciers/data/SurfaceVelocityData_utils.jl

```diff
@@ -0,0 +1,265 @@
+    nx, ny, ntimes = size(vx)
+    @assert length(date1) == length(date2) == ntimes
+    @assert nx == ny == 250
+
+    # Spatial preprocessing
+    params_projection = parse_proj(metadata(velRast)["proj4"])
+    hemisphere = (y[end]+y[begin])/2 >= 0 ? :north : :south
+    transform(X,Y) = Sleipnir.UTMercator(X, Y; zone=Int(params_projection["zone"]), hemisphere=hemisphere)
+    latitudes = map(x -> x.lat.val, transform.(mean(x), y))
+    longitudes = map(x -> x.lon.val, transform.(x, mean(y)))
+
+    if mapToGlacierGrid
+        # Here vx and vy are of type DiskArrays.BroadcastDiskArray
+        x, y, vx, vy = grid(glacier, latitudes, longitudes, vx, vy, mapping)
+        # Also retrieve the regional coordinates of the glacier grid through x and y
+        latitudes = glacier.Coords["lat"]
+        longitudes = glacier.Coords["lon"]
+        # Here vx and vy have been read from disk and they are arrays
+    else
+        # Access elements by converting a DiskArrays.BroadcastDiskArray to a real array
+        vx = vx[:,:,:]
+        vy = vy[:,:,:]
+    end
+
+    # Compute absolute velocity
+    vabs = (vx.^2 .+ vy.^2).^0.5
+    # The sqrt operation in Julia promotes Float32 to Float64. We convert manually
+    # to keep type consistency
+    vabs = convert(typeof(vx), vabs)
+
+    # The replace function promotes Float32 to Float64. We convert
+    # manually to keep consistency
+    vx = [eltype(x).(replace(vx[:,:,i], missing => NaN)) for i in 1:size(vx, 3)]
+    vy = [eltype(x).(replace(vy[:,:,i], missing => NaN)) for i in 1:size(vy, 3)]
+    vabs = [eltype(x).(replace(vabs[:,:,i], missing => NaN)) for i in 1:size(vabs, 3)]
+
+    if mapToGlacierGrid
+        # Set ice velocity to NaN outside of the glacier outlines
+        for i in range(1, size(vx,1))
+            vx[i][glacier.H₀ .== 0] .= NaN
+            vy[i][glacier.H₀ .== 0] .= NaN
+            vabs[i][glacier.H₀ .== 0] .= NaN
+        end
+    end
+
+    # Error is reported once per timespan, so upper bounds are given by absolute error
+    if !interp
+        # Since variables below are DiskArrays.BroadcastDiskArray, slice them to access their elements
+        vx_error = eltype(x).(replace(velRast.error_vx.data[:], missing => NaN))
+        vy_error = eltype(x).(replace(velRast.error_vy.data[:], missing => NaN))
+        # Absolute error uncertainty using propagation of uncertanties
+        if compute_vabs_error
+            vx_ratio_max = map(i -> ratio_max(vx[i], vabs[i]), 1:size(vx,1))
+            vy_ratio_max = map(i -> ratio_max(vy[i], vabs[i]), 1:size(vy,1))
+            vabs_error = ((vx_ratio_max .* vx_error).^2 .+ (vy_ratio_max .* vy_error).^2).^0.5
+            vabs_error = convert(typeof(vx_error[:]), vabs_error[:])
+        else
+            vabs_error = nothing
+        end
+    else
+        vx_error = nothing
+        vy_error = nothing
+        vabs_error = nothing
+    end
+
+    return SurfaceVelocityData(
+        x=x, y=y, lat=latitudes, lon=longitudes,
```

Comment:
I apologize in advance for the comment, but in terms of style I will try to always put spaces between assignment or binary operations (here `x = x` rather than `x=x`). If we decide to follow any code style in the future (which we can automatize, this will be done automatically) it will definivelly include this. At a personal level, it drives a bit crazy not to see spaces. I know I know, I am being annoying, but just a suggestion for the code in general.

Reply:
Sounds good!